src/main/scala/org/broadinstitute/dsde/workbench/avram/util/transformers.scala

```diff
@@ -0,0 +1,53 @@
+package org.broadinstitute.dsde.workbench.avram.util
+
+import cats.data.{EitherT, ReaderT}
+import cats.effect.IO
+import cats.implicits._
+import org.broadinstitute.dsde.workbench.avram.dependencies.{AvramDependencies, Global}
+import org.broadinstitute.dsde.workbench.avram.util.AvramError._
+
+import scala.concurrent.Future
+
+/**
+  * Defines an effect stack for Avram based on cats monad transformers. See:
+  * https://typelevel.org/cats/datatypes/eithert.html
+  * https://typelevel.org/cats/datatypes/kleisli.html
+  * https://typelevel.org/cats-effect/datatypes/io.html
+  */
+object transformers {
+  // AvramResult is a stack of the effects:
+  // - Reader
+  //   - IO
+  //     - Either
+  type AvramReader[A] = ReaderT[IO, AvramDependencies, A]
+  type AvramResult[A] = EitherT[AvramReader, AvramError, A]
+
+  // Conversions to AvramResult
+
+  def ioToResult[A](io: IO[A]): AvramResult[A] = {
+    EitherT.liftF(ReaderT.liftF(io))
+  }
+
+  def eitherToResult[A: AvramErrorable, B](either: Either[A, B]): AvramResult[B] = {
+    val withAvramError = either.leftMap(a => implicitly[AvramErrorable[A]].toAvramError(a))
+    EitherT.fromEither[AvramReader](withAvramError)
+  }
+
+  def withDependencies[A](reader: AvramDependencies => A): AvramResult[A] = {
+    withDependenciesIO(deps => IO.pure(reader(deps)))
+  }
+
+  def withDependenciesIO[A](readerIO: AvramDependencies => IO[A]): AvramResult[A] = {
+    EitherT.liftF(ReaderT(readerIO))
+  }
+
+  def futureToIO[A](future: Future[A]): IO[A] = {
+    IO.fromFuture(IO(future))
+  }
+
+  // AAAAHHHH
+  def unsafeRun[A](result: AvramResult[A], deps: AvramDependencies = Global.dependencies): A = {
+    result.value.run(deps).unsafeRunSync().fold(error => throw error.toServiceException, identity)
```

Comment:
Injects the dependency into `Reader`, runs the `IO`, and folds the `Either`

Reply:
Nit: I'd put this closer to the API endpoints, maybe in a supertype, to avoid the temptation to call it from anywhere else.

Reply:
That's a good idea

Reply:
Long-delayed follow-up: I also instinctively put `unsafeRun` here when applying this pattern. The other place I found I want to call `unsafeRun` is from tests. Hopefully, having "unsafe" in the name is enough of a warning to be careful when calling it.